### PR TITLE
Create pytorch_hub.md

### PR DESCRIPTION
**Pull request by jspisak**

### CODE REVIEW

File: _posts/pytorch_hub.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ ```
+
+Note that PyTorch Hub also allows auxillary entrypoints (other than pretrained models), e.g. ```bertTokenizer``` for preprocessing, to make the user workflow smoother.
+
+
+### Load a model
+
+Now that we know which models are available in the Hub, users can load a model entrypoint using the ```torch.hub.load()``` API. Because dependencies are taken care of by PyTorch Hub, this only requires a single command without the need to install a wheel.
+
+```python
```
Comment (ailzhang) [inline _posts/pytorch_hub.md]: In fact I realize we should add sth for `torch.hub.help()` here, like: 
`hub.help()` shows how to **instantiate** a model.
And add this line in the code block below.

```
print(torch.hub.help('huggingface/pytorch-pretrained-BERT', 'bertForMaskedLM'))
```

File: _posts/2019-6-5-towards-reproducible-research-with-pytorch-hub.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+**Q: Who hosts the model weights for PyTorch Hub?**
+A: You, as the contributor, are responsible to host the model weights. You can host your model in your favorite cloud storage or, if it fits within the limits, on GitHub.
+
+**Q: What if my model is trained on private data? Should I still contribute this model?**
+A: No! PyTorch Hub is centered around open research and that extends to the usage of open datasets to train these models on. If a pull request for a proprietary model is submitted, we will kindly ask that you resubmit a model trained on something open and available.
+
+**Q: Where are my downloaded models saved?**
+A: The locations are used in the order of:
+
+* Calling hub.set_dir(<PATH_TO_HUB_DIR>)
```
Comment (ailzhang) [inline _posts/2019-6-5-towards-reproducible-research-with-pytorch-hub.md]: * Calling `hub.set_dir()`
* `$TORCH_HOME/hub`, if environment variable `$TORCH_HOME` is set.
* `$XDG_CACHE_HOME/torch/hub`, if environment variable `$XDG_CACHE_HOME` is set.
* `~/.cache/torch/hub`

File: _posts/2019-6-5-towards-reproducible-research-with-pytorch-hub.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+### Load a model
+
+Now that we know which models are available in the Hub, users can load a model entrypoint using the ```torch.hub.load()``` API. This only requires a single command without the need to install a wheel. In addition the ```torch.hub.help()``` API can provide useful information about how to instantiate the model.
+
+```python
+print(torch.hub.help('huggingface/pytorch-pretrained-BERT', 'bertForMaskedLM'))
+model = torch.hub.load('huggingface/pytorch-pretrained-BERT', 'bertForMaskedLM', 'bert-base-cased')
+```
+
+In addition, it is also common that repo owners will want to continually add bug fixes or performance improvements. PyTorch Hub makes it super simple for users to get the latest update by calling:
```
Comment (ailzhang) [inline _posts/2019-6-5-towards-reproducible-research-with-pytorch-hub.md]: We have two "in addition" in a row" :P

File: _posts/2019-6-6-towards-reproducible-research-with-pytorch-hub.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+def bertForMaskedLM(*args, **kwargs):
+    """
+    BertForMaskedLM includes the BertModel Transformer followed by the
+    pre-trained masked language modeling head.
+    Example:
+      ...
+    """
+    model = BertForMaskedLM.from_pretrained(*args, **kwargs)
+    return model
+    ```
```
Comment (ailzhang) [inline _posts/2019-6-6-towards-reproducible-research-with-pytorch-hub.md]: Here format is off: we should delete the leading whitespace.

File: _posts/2019-6-6-towards-reproducible-research-with-pytorch-hub.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+```python
+>>> help(model.forward)
+>>>
+Help on method forward in module pytorch_pretrained_bert.modeling:
+forward(input_ids, token_type_ids=None, attention_mask=None, masked_lm_labels=None)
+...
+```
+
+Note that the ```*args```, ```**kwargs``` passed to hub.load() are used to *instantiate* a model.
```
Comment (ailzhang) [inline _posts/2019-6-6-towards-reproducible-research-with-pytorch-hub.md]: Here we are mainly talking about "running" a model, so we probably want to make this more clear

File: _posts/2019-6-6-towards-reproducible-research-with-pytorch-hub.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+* Go to [https://pytorch.org/hub](https://pytorch.org/hub) to learn more about the available models.
+
+
+A BIG thanks to the folks at HuggingFace, fast.ai and Nvidia as well as Morgane Riviere (FAIR Paris) and lots of others for helping bootstrap this effort!!
+
+
+## FAQ:
+
+**Q: If we would like to contribute a model that is already in the Hub but perhaps mine has better accuracy, should I still contribute?**
+A: Yes!! A next step for Hub is to implement an upvote/downvote system to surface the best models.
```
Comment (ailzhang) [inline _posts/2019-6-6-towards-reproducible-research-with-pytorch-hub.md]: W need extra line between "Q" and "A"s.

File: _posts/2019-6-6-towards-reproducible-research-with-pytorch-hub.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+**Q: Who hosts the model weights for PyTorch Hub?**
+A: You, as the contributor, are responsible to host the model weights. You can host your model in your favorite cloud storage or, if it fits within the limits, on GitHub. If it is not within your means to host the weights, check with us via opening an issue on the hub repository.
+
+**Q: What if my model is trained on private data? Should I still contribute this model?**
+A: No! PyTorch Hub is centered around open research and that extends to the usage of open datasets to train these models on. If a pull request for a proprietary model is submitted, we will kindly ask that you resubmit a model trained on something open and available.
+
+**Q: Where are my downloaded models saved?**
+
+We follow the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) and adhere to common standards around cached files and directories.
```
Comment (ailzhang) [inline _posts/2019-6-6-towards-reproducible-research-with-pytorch-hub.md]: This sentence should be merged into "A" ;)

**Comment by soumith** [comment]
merged
